ICSharpCode.Decompiler/IL/ControlFlow/AsyncAwaitDecompiler.cs

```diff
@@ -44,6 +45,11 @@ public static bool IsCompilerGeneratedStateMachine(Mono.Cecil.TypeDefinition typ
 			return false;
 		}
 
+		public static bool IsCompilerGeneratedMainMethod(MethodDefinition method)
+		{
+			return method.Name.Equals("<Main>", StringComparison.Ordinal);
```

Comment:
You should probably check, if the method is the entrypoint as well: `method.Module.Assembly?.EntryPoint == method`

Reply:
Got it, I added the check.